### PR TITLE
Group ViewEvents option and bug fix

### PR DESCRIPTION
**Pull request by lezzago**
### Description

Group ViewEvents option and bug fixes

### Issues Resolved

N/A

## Testing the changes

Manual testing with the Alerting and AD plugins

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx [hunk 1]
```diff
@@ -47,9 +47,14 @@ function getValueAxisPositions(embeddable: VisualizeEmbeddable): { left: boolean
...
  * @param savedObjectId the saved object id of the base vis
+ * @param embeddableStart Optional EmbeddableStart passed in for plugins to utilize the function
  */
-export async function fetchVisEmbeddable(savedObjectId: string): Promise<VisualizeEmbeddable> {
-  const embeddableVisFactory = getEmbeddable().getEmbeddableFactory('visualization');
+export async function fetchVisEmbeddable(
+  savedObjectId: string,
+  embeddableStart?: EmbeddableStart
+): Promise<VisualizeEmbeddable> {
+  const embeddableLoader = embeddableStart !== undefined ? embeddableStart : getEmbeddable();
```
Comment (ohltyler) [inline src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx:56]: Is this showing the same issues as UISettings etc. where it is not loaded when getting called by plugins?

File: src/plugins/vis_type_vislib/public/line_to_expression.ts [hunk 2]
```diff
@@ -32,7 +33,7 @@ export const toExpressionAst = async (vis: Vis, params: any) => {
   if (
     params.visLayers == null ||
     Object.keys(params.visLayers).length === 0 ||
-    !isEligibleForVisLayers(vis)
+    !isEligibleForVisLayers(vis, getUISettings())
```
Comment (ohltyler) [inline src/plugins/vis_type_vislib/public/line_to_expression.ts:36]: Is it required to make this call here? Or is this call here from vislib plugin also causing undefined issues? I was thinking this issue was only happening at external plugin fn calls.

File: src/plugins/vis_type_vislib/public/services.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@
 import { createGetterSetter } from '../../opensearch_dashboards_utils/public';
 import { DataPublicPluginStart } from '../../data/public';
 import { OpenSearchDashboardsLegacyStart } from '../../opensearch_dashboards_legacy/public';
+import { IUiSettingsClient } from 'opensearch-dashboards/public';
```
Comment (ohltyler) [inline src/plugins/vis_type_vislib/public/services.ts:34]: nit: linting error

File: src/plugins/vis_type_vislib/public/plugin.ts [hunk 4]
```diff
@@ -55,7 +55,7 @@ import {
 } from './vis_type_vislib_vis_types';
 import { ChartsPluginSetup } from '../../charts/public';
 import { DataPublicPluginStart } from '../../data/public';
-import { setFormatService, setDataActions, setOpenSearchDashboardsLegacy } from './services';
+import { setFormatService, setDataActions, setOpenSearchDashboardsLegacy, setUISettings } from './services';
```
Comment (ohltyler) [inline src/plugins/vis_type_vislib/public/plugin.ts:58]: linting error

File: src/plugins/visualizations/public/embeddable/visualize_embeddable.ts [hunk 5]
```diff
@@ -533,7 +533,7 @@ export class VisualizeEmbeddable
         this.savedAugmentVisLoader
       );
 
-      if (!isEmpty(augmentVisSavedObjs) && !aborted && isEligibleForVisLayers(this.vis)) {
+      if (!isEmpty(augmentVisSavedObjs) && !aborted && isEligibleForVisLayers(this.vis, getUISettings())) {
```
Comment (ohltyler) [inline src/plugins/visualizations/public/embeddable/visualize_embeddable.ts:536]: linting error

File: src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx (hunk 1 above)
Comment (lezzago) [reply src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx:56]: Yea I was having the same problem when calling it from the plugin (alerting/AD)

File: src/plugins/vis_type_vislib/public/line_to_expression.ts (hunk 2 above)
Comment (lezzago) [reply src/plugins/vis_type_vislib/public/line_to_expression.ts:36]: Let me test with removing this. I thought this would just be a safer bet mainly.

Comment (ohltyler) [reply src/plugins/vis_type_vislib/public/line_to_expression.ts:36]: Got it. IMO I'd like to try to keep existing working code in OSD as-is, and only have the workaround in the external plugins to ensure that we have the issue scoped correctly and that it is not also causing issues with other internal OSD plugins.

Reason being, so far it seems that this issue is only caused by some bootstrapping issue when building the compiled application where it is doing the extra installations on the AD/alerting plugins.

Comment (ohltyler) [reply src/plugins/vis_type_vislib/public/line_to_expression.ts:36]: And please update #4138 if you are seeing this coming from these internal plugins (vislib, visualizations, etc.).

**Review by ohltyler — APPROVED** [review]
LGTM!

File: src/plugins/vis_type_vislib/public/line_to_expression.ts (hunk 2 above)
Comment (lezzago) [reply src/plugins/vis_type_vislib/public/line_to_expression.ts]: It works with removing this, so updated the PR accordingly.

File: src/plugins/vis_augmenter/public/utils/utils.ts [hunk 6]
```diff
@@ -45,8 +45,9 @@ export const isEligibleForVisLayers = (vis: Vis): boolean => {
       (seriesParam: { type: string }) => seriesParam.type === 'line'
     ) &&
     vis.params?.type === 'line';
+
   // Checks if the augmentation setting is enabled
-  const config = getUISettings();
+  const config = uiSettingsClient !== undefined ? uiSettingsClient : getUISettings();
```
Comment (amsiglan) [inline src/plugins/vis_augmenter/public/utils/utils.ts:50]: Nit: use `uiSettingsClient ?? getUISettings()`

**Review by amsiglan — APPROVED** [review]

File: src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lezzago) [reply src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx]: fixed

File: src/plugins/vis_augmenter/public/utils/utils.ts (hunk 6 above)
Comment (lezzago) [reply src/plugins/vis_augmenter/public/utils/utils.ts]: fixed

**Pull request merged**